src/interp/mod.rs

```diff
@@ -212,6 +218,35 @@ impl LLVMIRInterpreter {
                         Name(n) => {
                             if n.as_str() == "printf" {
                                 self.printf(call);
+                            } else {
+                                match self.module.get_func_by_name(n) {
+                                    Some(func) => {
+                                        self.func_vars_stack.push(self.vars.clone());
+                                        self.vars.clear();
+
+                                        let func = func.clone();
+                                        for (par, arg) in
+                                            func.parameters.iter().zip(call.arguments.iter())
+                                        {
+                                            self.vars.insert(par.name.clone(), arg.0.clone());
+                                        }
+
+                                        let value = self.it_bb(&func);
```

Comment:
We have to be careful with recursion again here.

Reply:
But these are function calls and we want to return to the point where the function was called?

Reply:
We do. That doesn't mean it needs to be recursive though. We can just remember where we are before doing the call (by storing func, block, instruction index), and then continue interpreting from there after we've seen a return.